src/ProjectSystem/Impl/Extensions/ProjectTreeExtensions.cs

```diff
@@ -49,5 +49,16 @@ public static class ProjectTreeExtensions {
             }
             return Enumerable.Empty<string>();
         }
+
+        public static string GetSelectedFolderPath(this IImmutableSet<IProjectTree> nodes, UnconfiguredProject unconfiguredProject) {
+            if (nodes.Count == 1) {
+                var n = nodes.First();
+                if (n.Root == n) {
```

Comment:
Not really, but they have an extension :)
```{csharp}
namespace Microsoft.VisualStudio.ProjectSystem
{
    public static class ProjectTreeExtensions
    {
        /// <summary>
        /// Checks whether the given node is a root node in its tree.
        /// </summary>
        /// <param name="node">The node to test.</param>
        /// <returns><c>true</c> if the node is the root node; otherwise <c>false</c>.</returns>
        [Pure]
        public static bool IsRoot(this IProjectTree node)
        {
            Contract.Requires(node != null);
 
            Requires.NotNull(node, nameof(node));
            return node.Root == node;
        }
 
```